pkg/controller/sync/updater.go

```diff
@@ -117,15 +115,29 @@ func (u *federatedUpdaterImpl) Create(clusterName string) {
 			return "", err
 		}
 		createdObj, err := client.Resources(obj.GetNamespace()).Create(obj, metav1.CreateOptions{})
-		if err != nil {
+		if err == nil {
+			return util.ObjectVersion(createdObj), nil
+		}
+		// TODO(marun) Figure out why attempting to create a namespace that
+		// already exists indicates ServerTimeout instead of AlreadyExists.
+		alreadyExists := apierrors.IsAlreadyExists(err) || u.fedResource.TargetKind() == util.NamespaceKind && apierrors.IsServerTimeout(err)
+		if !alreadyExists {
 			return "", err
 		}
-		return util.ObjectVersion(createdObj), err
+
+		// Attempt to update the existing resource to ensure that it
+		// is labeled as a managed resource.
+		clusterObj, err := client.Resources(obj.GetNamespace()).Get(obj.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return "", errors.Wrapf(err, "Failed to retrieve object potentially requiring adoption for cluster %q", clusterName)
+		}
+		u.Update(clusterName, clusterObj)
```

Comment:
I think it should be possible for the update on a pre-existing object to also be ignored based on user pre-set configuration. I think a safer option would be that the object is not updated if the configuration is missing or not set.

Reply:
I agree that configurable adoption is not in scope for this PR.  

I think adoption should remain the default, and I am also supportive of your adding a flag (er, `FederationConfig` option) that would ensure that adoption would never occur (i.e. an existing resource in a member cluster would only become managed by federation if a user manually set the label `federation.k8s.io/managed: true`).